interfaces/repo.go

```diff
@@ -516,3 +516,101 @@ func (r *Repository) securitySnippetsForSnap(snapName string, securitySystem Sec
 	}
 	return snippets, nil
 }
+
+// AddSnap adds plugs and slots declared by the given snap to the repository.
+//
+// This function can be used to implement snap install or, when used along with
+// RemoveSnap, snap upgrade.
+//
+// Plugs/slots that are already present in the repository are not changed. Only
```

Comment:
Given how ``RemoveSnap`` works I wonder if this should just panic. The intent is for those to be called in a given state so that they can just do their job and not handle anything non-trivial.